server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1088,8 +1188,75 @@ protected void parseCreateField(ParseContext context) throws IOException {
         }
     }
 
+    @Override
+    public CheckedConsumer<IndexTimeScriptContext, IOException> getPostParsePhase() {
+        if (this.script == null) {
+            return null;
+        }
+        return c -> this.script.compiledScript.executeAndIndex(
+            c.searchLookup,
+            c.leafReaderContext,
+            0,
+            v -> this.indexValue(c.pc, v)
+        );
+    }
+
     @Override
     public FieldMapper.Builder getMergeBuilder() {
         return new Builder(simpleName(), type, ignoreMalformedByDefault, coerceByDefault).init(this);
     }
+
+    private interface NumberMapperScript {
```

Comment:
just by looking at it, it is hard to tell why this class is needed. I think it holds a compiled script, as scripts get compiled when the mappings are parsed. Maybe we should better define what abstractions are needed: it seems that Script is not enough as it is not compiled, and we need some representation for a compiled script, that then gets executed once for each document being indexed. I even wonder if this should be an interface or rather a base class, because there are aspects of a compiled script that all of the types share (params, fieldName)?

Reply:
We need to make a distinction between a `Script`, which just holds the json representation of a script (so is needed for serialization and comparison during merges, etc); then the compiled script (here either `LongFieldScript` or `DoubleFieldScript`) which is shared with runtime fields; then this mapper script, which executes the compiled script, retrieves its values, and passes it to the indexer.  This interface is the last one, and will probably end up being public and shared - maybe I should look at adding a KeywordFieldMapper implementation here as well, as that should confirm which of these interfaces will be common between different mapper types?